core/src/main/java/hudson/model/Slave.java

```diff
@@ -373,7 +375,7 @@ private URLConnection connect() throws IOException {
             return res.openConnection();
         }
 
-        public URL getURL() throws MalformedURLException {
+        public URL getURL() throws IOException {
```

Comment:
@jglick It is binary compatible iff the WAR is exploded correctly. Custom WARs may really cause unhandled IOExceptions